pkgs/native_assets_builder/test_data/complex_link/hook/build.dart

```diff
@@ -2,21 +2,35 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:io';
+
 import 'package:native_assets_cli/native_assets_cli.dart';
 
-const packageName = 'complex_link';
+void main(List<String> args) async {
+  await build(args, (config, output) async {
+    final packageName = config.packageName;
+    final assetDirectory =
+        Directory.fromUri(config.packageRoot.resolve('assets/'));
+    // If assets are added, rerun hook.
```

Comment:
Why rerun whenever any files are added to the `assets` folder?

Reply:
Because I'm assuming the typical user will just drop a file in the assets dir, and then expect that if they restart their app (or hot restart in the future) the asset will show up!

Reply:
Probably this is such a common use case that we should even add a helper function to just designate a dir as data assets dir. Some questions for such a helper function:

* Should it maybe not drop the file extension.
* Should it recurse into subdirectories?
* Should it have the directory name itself as part of the asset id? e.g. the asset ID would simply be `package:my_package/assets/foo.json`.